server/services/echo_service_test.go

```diff
@@ -367,7 +367,7 @@ func (m *mockChatStream) SetTrailer(md metadata.MD) {
 }
 
 func (m *mockChatStream) verify(expectHeadersAndTrailers bool) {
-	if expectHeadersAndTrailers && !reflect.DeepEqual([]string{"show", "case"}, m.trail) && !reflect.DeepEqual([]string{"showcaseHeader", "anotherHeader"}, m.head) {
+	if expectHeadersAndTrailers && (!reflect.DeepEqual([]string{"show", "case"}, m.trail) || !reflect.DeepEqual([]string{"showcaseHeader", "anotherHeader", "showcaseHeader", "anotherHeader"}, m.head)) {
```

Comment:
It's not clear if the additional `"showcaseHeader", "anotherHeader"` is another bug appearing as a result of fixing the test